src/Pos/Lrc/Worker.hs

```diff
@@ -142,23 +145,23 @@ lrcDo getPeers epoch consumers tip = tip <$ do
         Nothing -> throwM UnknownBlocksForLrc
         Just (NewestFirst -> blunds) ->
             withBlocksRolledBack blunds $ do
-                -- We have to compute richmen before calculating the seed because
-                -- seed calculation relies on knowing richmen
                 issuersComputationDo epoch
-                richmenComputationDo epoch consumers
+                -- We have to compute richmen before calculating the seed
+                -- because seed calculation relies on knowing richmen. We
+                -- give 'calcAndStoreSeed' as a callback to
+                -- 'richmenComputationDo' because it's easier than reading
+                -- richmen from the database (I couldn't find a function that
+                -- does it on a locked LRC database)
+                let simpleConsumer = lrcConsumerFromComponentSimple @RCSsc bvdMpcThd
+                let calcSeedConsumer = simpleConsumer {
+                        lcComputedCallback = calcAndStoreSeed }
+                richmenComputationDo epoch (calcSeedConsumer:consumers)
```

Comment:
“richmen from 9th epoch” means “richmen computed by LRC when 9 was passed as argument”. It uses distribution from `crucialSlot 9`, i. e. `{8, epochSlots - slotSecurityParam - 1}`.

When we compute seed for the 10-th epoch we want to use richmen from `lrc 9`.

When we launch LRC for `9`-th epoch we use stake distrubtion